### PR TITLE
Compass 3.0.0

### PR DESCRIPTION
**Pull request by vadymmarkov**
@zenangst @RamonGilabert @onmyway133 
We kinda like to break `Compass` from time to time, so it's a brand new breaking change 😄 

1) Remove `Compass.parse` completion closure in favour of `Location` struct:

So instead of:

``` swift
func application(app: UIApplication,
  openURL url: NSURL,
  options: [String : AnyObject]) -> Bool {
    return Compass.parse(url) { route, arguments in
      switch route {
        case "profile:{username}":
          let profileController = profileController(title: arguments["{username}"])
          self.navigationController?.pushViewController(profileController, animated: true)
        default: break
      }
    }
}
```

We unwrap optional `Location`:

``` swift
func application(app: UIApplication,
  openURL url: NSURL,
  options: [String : AnyObject]) -> Bool {
    guard let location = Compass.parse(url) else {
      return false
    }

    let arguments = location.arguments

    switch location.path {
      case "profile:{username}":
        let profileController = profileController(title: arguments["{username}"])
        self.navigationController?.pushViewController(profileController, animated: true)
      default: break
    }

    return true
}
```

The advantage of this approach is that `Compass.parse` is not async anymore, we get the needed result straight away and can handle the case when `urn` could not be parsed.

2) Refactor `Router` and `Routable` to support Mac by using type aliases.
3) Route path, arguments and fragments are scoped by `Location` struct. Makes it easier to add more properties in the future.
4) We used only 2 `String` methods from `Sugar` and I feel like it's unnecessary to hold this dependency, especially thinking about Swift 3 where it's no longer actual. So... we don't use `Sugar` in `Compass` anymore 😄 
5) Refactor project structure to be prepared to Swift Package Manager.
6) Add Swiftlint, mention-bot and update Travis config.

### CODE REVIEW

**Comment by jessearmand** [comment]
I agree with those ideas. I have removed Sugar as a dependency as well in my own fork.

File: Compass.podspec [hunk 1]
```diff
@@ -13,7 +13,7 @@ Pod::Spec.new do |s|
 
   s.requires_arc = true
 
-  s.ios.source_files = 'Sources/{iOS,Shared}/**/*'
-  s.osx.source_files = 'Sources/{Mac,Shared}/**/*'
-  s.dependency 'Sugar'
+  s.ios.source_files = 'Sources/**/*'
```
Comment (onmyway133) [inline Compass.podspec]: I think we can reduce to just `s.source_files`

File: Sources/Location.swift [hunk 2]
```diff
@@ -0,0 +1,16 @@
+public struct Location {
```
Comment (onmyway133) [inline Sources/Location.swift:1]: This is nice

File: Compass.podspec (hunk 1 above)
Comment (vadymmarkov) [reply Compass.podspec]: 👍

**Comment by vadymmarkov** [comment]
Well, seems like it's gonna be v3 because the last released is v2. We just had a wrong one in Podspec.

**Comment by onmyway133** [comment]
🎸 🎶 🎶

**Comment by zenangst** [comment]
This is grrreeeeat!

young-shaq-dancing-shaq-gifs

**Pull request merged**